gossip/src/crds.rs

```diff
@@ -56,6 +57,26 @@ use {
 const CRDS_SHARDS_BITS: u32 = 12;
 // Number of vote slots to track in an lru-cache for metrics.
 const VOTE_SLOTS_METRICS_CAP: usize = 100;
+// Number of ending zero bits for crds signature to get reported to influx
+// mean new push messages received per minute per node
+//      testnet: ~500k,
+//      mainnet: ~280k
+// target: 1-2 signatures reported per minute
+// log2(250k) = ~17.9.
+const SIGNATURE_SAMPLE_TRAILING_ZEROS: u8 = 18;
+
+pub fn trailing_n_crds_signature_bits_are_zero(signature_bytes: &[u8], n: u8) -> bool {
+    let num_bytes: usize = (n as usize + 7) / 8; // Number of bytes required to represent n bits
+
+    let mut signature_ending: u64 = 0;
+    for (i, signature_byte) in signature_bytes.iter().enumerate().take(num_bytes) {
+        let shift = i * 8;
+        signature_ending |= (*signature_byte as u64) << shift;
+    }
+
+    // check if last n bits of signature (aka signature_ending) are all 0.
+    (signature_ending & ((1 << n) - 1)) == 0
```

Comment:
Instead of this we can convert the first 8 bytes to a u64 using `from_le_bytes`:
https://doc.rust-lang.org/std/primitive.u64.html#method.from_le_bytes

and then just use `trailing_zeros`:
https://doc.rust-lang.org/std/primitive.u64.html#method.trailing_zeros